### PR TITLE
[opentelemetry-collector] Add support to generate CRD

### PR DESCRIPTION
**Pull request by hivp**
### Description

This PR adds support to generate the CRD (`kind: OpenTelemetryCollector`) directly from `opentelemetry-collector` chart and use it to deploy so `opentelemetry-operator` is able to detect that deployment an create the proper type (daemonset, deployment, statefulset).

### CODE REVIEW

File: charts/opentelemetry-collector/values.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/values.yaml:9]: This should be false by default.

File: charts/opentelemetry-collector/templates/cr-collector.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/templates/cr-collector.yaml:16]: To be fully configurable we need to allow `mode` to be `sidecar`.  We should fail in `NOTES.txt` if `mode=sidecar` and `defineCustomResource=false`

File: charts/opentelemetry-collector/templates/_config.tpl
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/templates/_config.tpl:86]: Do we need this?  The `mode=statefulset` configuration uses the deployment config since deployments and statefulsets don't need different collector configuration.

**Review by TylerHelmuth — COMMENTED** [review]
@hivp this is a pretty exciting addition, but we need to have a discussion about it as there has been lots of discussion in the past on how to do this (open issue here: https://github.com/open-telemetry/opentelemetry-helm-charts/issues/69).  

I think the big question is whether or not we need to have a dedicated chart or the operator's custom resources, the operator chart should be about to create them, or if this chart should be the one to handle `OpenTelemetryCollector` and the `Instrumentation` resource should be handled somewhere else.

I left some review comments that will be beneficial if we chose to move forward with this approach vs a dedicated.

@open-telemetry/helm-approvers please review.

File: charts/opentelemetry-collector/values.yaml
Comment (hivp) [reply charts/opentelemetry-collector/values.yaml]: Resolved:
https://github.com/nhuray/opentelemetry-helm-charts/commit/28a2ad2d4c46f9808a2102b80a45f84cd92f3c3d

File: charts/opentelemetry-collector/templates/cr-collector.yaml
Comment (hivp) [reply charts/opentelemetry-collector/templates/cr-collector.yaml]: Resolved:
https://github.com/nhuray/opentelemetry-helm-charts/commit/28a2ad2d4c46f9808a2102b80a45f84cd92f3c3d

File: charts/opentelemetry-collector/templates/_config.tpl
Comment (hivp) [reply charts/opentelemetry-collector/templates/_config.tpl]: Resolved:
https://github.com/nhuray/opentelemetry-helm-charts/commit/28a2ad2d4c46f9808a2102b80a45f84cd92f3c3d

File: charts/opentelemetry-collector/templates/_helpers.tpl
Comment (Allex1) [inline charts/opentelemetry-collector/templates/_helpers.tpl:147]: same as https://github.com/open-telemetry/opentelemetry-helm-charts/pull/748/files#r1170208321

**Comment by Allex1** [comment]
@hivp Thanks for this contribution. This is very exciting. I would support adding this functionality to the collector repo and reuse the existing code.
Before we can further discuss/review/test can you please follow our [contribution guide](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/CONTRIBUTING.md) in order to have CI pass and maybe add support for sidecar as well.

File: charts/opentelemetry-collector/values.yaml
Comment (Allex1) [inline charts/opentelemetry-collector/values.yaml:12]: same as https://github.com/open-telemetry/opentelemetry-helm-charts/pull/748/files#r1170208321

**Comment by hivp** [comment]
Hey guys:
I have added support for sidecar: https://github.com/nhuray/opentelemetry-helm-charts/commit/1ea8f05d3ac04e8448c5dd48ff1db43edf5a74e1 . I tested it in my k8s cluster and worked.



And some other changes according to your [contribution](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/CONTRIBUTING.md) guide:  https://github.com/nhuray/opentelemetry-helm-charts/commit/38b74cdf7f7af2d5481a877b02bea4b0ad057965 , https://github.com/nhuray/opentelemetry-helm-charts/commit/5644959c796a81c58768c53f4f7cd884cef685c4

**Comment by TylerHelmuth** [comment]
> reuse the existing code

@Allex1 can you expand on this?  Do you mean if we wanted to install the custom resource as part of the operator chart or dedicated chart we'd reuse this chart as a subchart?

**Comment by Allex1** [comment]
> > reuse the existing code
> 
> @Allex1 can you expand on this? Do you mean if we wanted to install the custom resource as part of the operator chart or dedicated chart we'd reuse this chart as a subchart?

@TylerHelmuth I'm torn between having the CR in the collector chart or bundled in the operator chart like kps. 
I think because kps is focused only on monitoring it makese sense to have cr's bundled together with the operator.
Currently we deploy the otel operator chart once per cluster and the collector multiple times per use case so I'm leaning more towards having the cr in the collector chart.

**Comment by dmitryax** [comment]
Hi folks, can we have an issue to discuss it first before considering merging this? (as we have been doing before) I don't understand why CRD should be part of this chart. It doesn't work without operator, and this chart should not depend on installation of the operator.

**Review by dmitryax — CHANGES_REQUESTED** [review]
At this point, I don't think CRD should be part of this chart. Let's discuss it in an issue first

**Pull request closed without merging**
